code/model/encoders/LSTM_encoder.py

```diff
@@ -3,27 +3,26 @@
 import numpy as np
 from torch.nn.utils.rnn import pack_padded_sequence,pad_packed_sequence,pad_sequence
 
-
 class LSTM_encoder(torch.nn.Module):
-    def __init__(self,embed_size,hidden_size,num_layers,vocab,batch_size,sorted_seq=True,
-                 bidirectional=False):
+    def __init__(self,embed_layer,vocab,config):
         '''
         :param embed_size:  Input size of LSTM
+        :param embed_layer: embedding layer
         :param hidden_size: Hidden size of LSTM
         :param num_layers: Number of recurrent layers.
         :param vocab: Vocabulary of embedding
-        :param batch_size: Batch size
+        :param dropout: dropout layer follows lstm layer
         :param sorted_seq: If the input_seq has been sorted,set this param to True and give
                            the seq and lengths when forward function is called. Default: False
         :param bidirectional: If True, becomes a bidirectional LSTM. Default: False
         '''
         super(LSTM_encoder,self).__init__()
-        self.batch_size = batch_size
-        self.sorted_seq = sorted_seq
+        self.sorted_seq = config.enc_sorted_seq
         self.pad = vocab.pad
-        n_vocab = len(vocab.i2c)
-        self.embedding = nn.Embedding(n_vocab,embed_size,padding_idx=self.pad)
-        self.lstm = nn.LSTM(embed_size,hidden_size,num_layers,bidirectional = bidirectional)
+        self.embedding = embed_layer
+        self.lstm = nn.LSTM(config.embed_size,config.enc_hidden_size,
+                            config.enc_num_layers,bidirectional = config.enc_bidirectional,
+                            dropout= config.enc_dropout if config.enc_num_layers > 1 else 0)
 
     def forward(self,seq,lengths=None):
```

Comment:
dataloader得到的一个batch = (padded_mols: torch.tensor, lengths: list)
建议直接输入，然后内部再拆分
```suggestion
    def forward(self, x):
        seqs, lengths = x
```

Reply:
这个LSTM_encoder写的时候是按没排序写的，输入可以是一个单独的tensor，所以这里分开写了；不过在model.py里面，forward部分是直接给batch就行，之后再内部拆分